consensus/cryptarchia-engine/src/lib.rs

```diff
@@ -390,4 +388,94 @@ pub mod tests {
         }
         assert_eq!(engine.tip_id(), parent);
     }
+
+    #[test]
+    fn test_orphan_proof_import() {
+        let coin = Coin::new(0);
+        let mut engine = engine(&[coin.commitment()]);
+
+        let coin_new = coin.evolve();
+        let coin_new_new = coin_new.evolve();
+
+        // produce a fork where the coin has been spent twice
+        let fork_1 = block(1, *engine.genesis(), coin);
+        let fork_2 = block(2, fork_1.header().id(), coin_new);
+
+        // neither of the evolved coins should be usable right away in another branch
+        assert!(matches!(
+            engine.receive_block(block(1, *engine.genesis(), coin_new)),
+            Err(Error::LedgerError(LedgerError::CommitmentNotFound))
+        ));
+        assert!(matches!(
+            engine.receive_block(block(1, *engine.genesis(), coin_new_new)),
+            Err(Error::LedgerError(LedgerError::CommitmentNotFound))
+        ));
+
+        // they also should not be accepted if the fork from where they have been imported has not been seen already
+        assert!(matches!(
+            engine.receive_block(block_with_orphans(
+                1,
+                *engine.genesis(),
+                coin_new,
+                vec![fork_1.header().clone()]
+            )),
```

Comment:
isn't this clearer?
```suggestion
            engine.receive_block(
              block(1, *engine.genesis(), coin_new)
                .with_orphans(vec![fork_1.header().clone()])
            ),
```

Reply:
I would prefer not to provide an api to change the header of a block after it has been created